Formula/libxml2.rb

```diff
@@ -41,7 +41,7 @@ class Libxml2 < Formula
 
   keg_only :provided_by_osx
 
-  depends_on :python => :optional
+  depends_on :python if MacOS.version <= :snow_leopard
```

Comment:
At this point I think the only solution that is going to keep most people happen making `python` keg-only and making everything using python bindings depend on it. The problem people have with non-system Python (and indeed non-system anything) being pulled in isn't (amongst sane people) the bandwidth/space but that ` which python` goes from the system Python to Homebrew's python without it being their intention. Yes, this may expose some "I `brew install`d foo and it's not in my PATH!" pain but that's something we may need to fix, too.